### PR TITLE
Add CSI docs

### PR DESCRIPTION
**Pull request by darkowlzz**

### CODE REVIEW

**Comment by darkowlzz** [comment]
Thanks!
Added an entry in _guides/002-install_and_run.md, generated the docs and it shows up in the menu now.

File: _guides/002-install_and_run.md [hunk 1]
```diff
@@ -63,6 +63,10 @@ sections:
         title: Dynamic provisioning
         description: Dynamic provisioning
         module: install/kubernetes/dynamic-provisioning
+      -
+        title: Dynamic provisioning (CSI)
+        description: Dynamic provisioning (CSI)
+        module: install/kubernetes/dynamic-provisioning-with-csi
```
Comment (oicheryl) [inline _guides/002-install_and_run.md]: This hierarchy doesn't feel right. I'd prefer a Kubernetes (CSI) item at the same level as Kubernetes.

**Comment by darkowlzz** [comment]
Created a new section Kubernetes (CSI) with a little modified index.md, removing prerequisites that are no longer required for CSI installation with k8s 1.10.

File: _docs/install/kubernetes-csi/dynamic-provisioning.md [hunk 2]
```diff
@@ -0,0 +1,281 @@
...
+removes the need to explicitly create secrets for kubernetes to talk to
+StorageOS driver.
+
+StorageOS volumes can be created on-demand through dynamic provisioning.
+
+1. Adminstrators create storage classes to define different types of storage.
+1. Users create a persistent volume claim (PVC).
+1. The user references the PVC in a pod.
+
+The [StorageOS Helm chart for CSI deployment](https://github.com/storageos/helm-chart/tree/csi-deployment) includes the
```
Comment (oicheryl) [inline _docs/install/kubernetes-csi/dynamic-provisioning.md]: NB we should have one helm chart that can deploy both CSI and non-CSI, controlled by an option.

**Pull request merged**